src/lib/modules/blockchain_connector/index.js

```diff
@@ -24,6 +25,16 @@ class BlockchainConnector {
     this.wait = options.wait;
     this.contractsSubscriptions = [];
     this.contractsEvents = [];
+    this.logFile = fs.dappPath(".embark", "contractEvents.json");
+
+    this.writeLogFile = async.queue(({data}, callback) => {
+      fs.writeJson(this.logFile, data, err => {
+        if (err) {
+          console.error(err);
+        }
+        callback();
+      });
+    }, 1);
```

Comment:
Problem with the queue is you can only have one job done at a time. So if 5 events are triggered in fast succession, you will have to write 5 times one after the other. (including 5 reads)

Whereas, with the cargo, you will receive an array of tasks (the 4/5 json datas) that you can iterate over before actually writing. Example:
```
    this.writeLogFile = async.cargo((tasks, callback) => {
      const data = this._readEvents();

      tasks.forEach(task => {
          data[new Date().getTime()] = task;
      });

      fs.writeJson(this.logFile, data, err => {
        if (err) {
          console.error(err);
        }
        callback();
      });
    }); // Put here no limit, as we can handle all tasks
```

Reply:
TIL 💯 will make the change

Reply:
I have a createWriteStream variant almost ready too, after the exchange of comments yesterday I decided to give it a shot.